lib/bowling_game.rb

```diff
@@ -0,0 +1,104 @@
+require_relative 'frame'
+
+class BowlingGame
+  attr_reader :rolls
+  def initialize
+    @score, @roll_number = 0, 0
+    @rolls = []
+  end
+
+  def bowl(pins, frame = Frame)
+    raise "The game is over." if game_over
+    if regular_strike(pins)
+      create_strike_frame(pins, frame)
+    elsif final_frame?
+      create_bonus_frame(pins)
+    elsif first_roll?
+      create_new_frame(pins, frame)
+    else
+      close_frame(pins, frame)
+    end
+  end
+
+  def score
+    calculate_score
+    @score
```

Comment:
Does `@score` need to be an instance variable which is stored within the Bowling Game class state. Could we instead calculate and return the result to the user without using an instance variable?

Storing it as state causes issues with the current code:
```
>bg = BowlingGame.new
>bg.bowl(3)
> bg.bowl(6)
> bg.score
 => 9
> bg.score
 => 18
```

As you can see in this example (which I copied from irb), by always doing += on the stored state, we'll keep incrementing the score instance variable each time we call `.score`. If we instead calculated and returned the score without storing it, we wouldn't have this problem because after we returned the local variable it would go out of scope and become inaccessible next time we called `.score`.